lib/class-wp-json-posts-controller.php

```diff
@@ -106,4 +869,196 @@ public function prepare_item_for_response( $post, $request ) {
+			),
+			'collection'      => array(
+				'href' => json_url( '/wp/posts' ),
+			),
+			'version-history' => array(
+				'href' => json_url( '/wp/posts/' . $post->ID . '/revisions' ),
+			),
+		);
+
+		if ( ! empty( $schema['properties']['author_id'] ) ) {
+			$links['author']  = array(
+				'href' => json_url( '/wp/users/' . $post->post_author ),
+				'embeddable' => true,
+			);
+		}
+
+		if ( ! empty( $schema['properties']['comments'] ) ) {
+			$links['comments'] = array(
+				'href' => json_url( '/wp/posts/' . $post->ID . '/comments' ),
+			);
+		}
+
+		if ( ! empty( $schema['properties']['parent_id'] ) && ! empty( $post->post_parent ) ) {
+			$links['up'] = array(
+				'href' => json_url( '/wp/posts/' . (int) $post->post_parent ),
+				'embeddable' => true,
+			);
+		}
+
+		return $links;
+	}
+
+	/**
+	 * Get the schema for a Post
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'title'                => 'Post',
+			'type'                 => 'object',
+			/*
+			 * Base properties for every Post
+			 */
+			'properties'           => array(
+				'id'               => array(
+					'description'  => 'The unique identifier for a Post.',
+					'type'         => 'integer',
+				),
+				'type'             => array(
+					'description'  => 'An alphanumeric identifier for a Post unique to its type.',
+					'type'         => 'string',
+				),
+				'slug'             => array(
+					'description'  => 'An alphanumeric identifier for a Post unique to its type.',
+					'type'         => 'string',
+				),
+				'link'             => array(
+					'description'  => 'URL to the Post.',
+					'type'         => 'string',
+				),
+				'published_date'   => array(
+					'description'  => 'The date the Post was published.',
+					'type'         => 'string',
+					'format'       => 'date-time',
+				),
+				'modified_date'    => array(
+					'description'  => 'The date the Post was last modified.',
+					'type'         => 'string',
+					'format'       => 'date-time',
+				),
+			),
+		);
+
+		$post_type_obj = get_post_type_object( $this->post_type );
+		if ( $post_type_obj->hierarchical ) {
+			$schema['properties']['parent_id'] = array(
+				'description'      => 'The ID for the parent of the Post.',
+				'type'             => 'integer',
+				);
+		}
+
+		if ( 'post' === $this->post_type ) {
+			$schema['properties']['sticky'] = array(
+				'description'      => 'Whether or not the Post should be treated as sticky.',
+				'type'             => 'bool',
+				);
+		}
+
+		$post_type_attributes = array(
+			'title',
+			'editor',
+			'author',
+			'excerpt',
+			'thumbnail',
+			'comments',
+			'revisions',
+			'page-attributes',
+			'post-formats',
+			);
+		foreach( $post_type_attributes as $attribute ) {
+			if ( ! post_type_supports( $this->post_type, $attribute ) ) {
+				continue;
+			}
+
+			switch( $attribute ) {
+
+				case 'title':
+					$schema['properties']['title'] = array(
```

Comment:
@danielbachhuber Shouldn't 'title', 'excerpt', 'content', and 'guid' be objects, with 'rendered' and 'raw' properties?